zoobc/zoobc-core
# Make sure db tx complete in push block

## Description
Refactoring PushBlock function to prevent DB transaction to slip away without commit or rollback.

## Breakdown
- [x] Refactoring PushBlock in `blockMainService`
- [x] Refactoring PushBlock in `blockSpineService`

## Reference Issue
Close #numberIssue

## Step to Test (optional)
- run the node locally
- make sure the mainchain and spinechain correctly smithing
<img width="1440" alt="Screen Shot 2021-01-07 at 10 24 00" src="https://user-images.githubusercontent.com/17913266/103843574-0d83e000-50d3-11eb-8a1f-07e0b0d95ee6.png">
